autoPyTorch/utils/results_manager.py

```diff
@@ -413,11 +423,31 @@ def _extract_results_from_run_history(self, run_history: RunHistory) -> None:
             config = run_history.ids_config[run_key.config_id]
             self._update(config=config, run_key=run_key, run_value=run_value)
 
+        self._check_null_in_optional_inference_choices()
+
         self.rank_opt_scores = scipy.stats.rankdata(
             -1 * self._metric._sign * self.opt_scores,  # rank order
             method='min'
         )
 
+    def _check_null_in_optional_inference_choices(
+        self
+    ) -> None:
+        """
+        Checks if the data is missing for each optional inference choice and
+        sets the scores for that inference choice to all None.
```

Comment:
It would be nice to also add the case of `score == metric._worst_possible_result` in the docstring:

`Checks if the data is missing or if all scores are equal to the worst possible result for each optional inference choice and sets the scores for that inference choice to all None.`